server/app/assets/javascripts/admin_validation.ts

```diff
@@ -76,6 +76,30 @@ class AdminValidationController {
     return multiOptionIsValid
   }
 
+  /**
+   * Validates that there are no invalid admin names and updates the field error state.
+   * @param {HTMLInputElement[]} inputElements All the inputElements to verify
+   * @param {string} errorClass The error class to show on error
+   *
+   * @return {boolean} true if all fields are valid
+   *  */
+  private validateMultiOptionQuestionAdminNames(
+    inputElements: HTMLInputElement[],
+    errorClass: string,
+  ) {
+    let multiOptionIsValid = true
+    for (const inputElement of inputElements) {
+      // Assert that the admin name is not blank, and that it only
+      // contains 0-9, a-z, A-Z, _, and -
+      const inputIsValid = /^[0-9a-zA-Z_-]+$/.test(inputElement.value)
+      this.updateFieldErrorState(inputElement, errorClass, inputIsValid)
+      if (!inputIsValid) {
+        multiOptionIsValid = inputIsValid
+      }
+    }
+    return multiOptionIsValid
```

Comment:
suggestion (non-blocking): could do?
```suggestion
    for (const inputElement of inputElements) {
      // Assert that the admin name is not blank, and that it only
      // contains 0-9, a-z, A-Z, _, and -
      const inputIsValid = /^[0-9a-zA-Z_-]+$/.test(inputElement.value)
      this.updateFieldErrorState(inputElement, errorClass, inputIsValid)
      
      if (!inputIsValid) {
        return false
      }
    }
    
    return true
```

Reply:
We need to visit every option admin name to decide if its error message should be shown. If we exit on the first failure, we won't visit them all.

Reply:
oh, woops. sorry!